### PR TITLE
Xen 4.8 support

### PR DESCRIPTION
**Pull request by tklengyel**
Xen 4.8 features this PR adds support to:
- ARM registers being sent via vm_event
- Instruction emulator buffer returned for int3 events

### CODE REVIEW

**Comment by tklengyel** [comment]
PR tested with both Xen 4.7 and 4.8-rc, works as expected.

**Comment by tklengyel** [comment]
Retest this please.

**Comment by tklengyel** [comment]
Retest this please.

File: libvmi/debug.h [hunk 1]
```diff
@@ -53,6 +53,6 @@ typedef enum {
 } vmi_debug_flag_t;
 
 /* uncomment this and recompile to enable debug output */
-#define VMI_DEBUG VMI_DEBUG_EVENTS
```
Comment (tklengyel) [inline libvmi/debug.h]: This seems to have crept into the master branch from some other PR..

Comment (tklengyel) [inline libvmi/debug.h]: Check the next line that's being added.

Comment (smaresca) [inline libvmi/debug.h]: Oh I see. Displayed strangely, only showed the - without the +. Nevermind :)

**Pull request merged**